cpp/include/cudf/detail/iterator.cuh

```diff
@@ -134,6 +232,58 @@ auto make_null_replacement_iterator(column_device_view const& column,
       null_replaced_value_accessor<Element>{column, null_replacement});
 }
 
+/**
+ * @brief Constructs a pair iterator over a column's values that replaces null
+ * elements with a specified value and its validity.
+ *
+ * Dereferencing the returned iterator for element `i` will return `pair(column[i], true)`
+ * if it is valid, or `pair(null_replacement, false)` if it is null.
+ * This iterator is only allowed for nullable columns.
+ *
+ * @throws `cudf::logic_error` if the column is not nullable.
+ * @throws `cudf::logic_error` if column datatype and Element type mismatch.
+ *
+ * @tparam Element The type of elements in the column
+ * @param column The column to iterate
+ * @param null_replacement The value to return for null elements
+ * @return auto Iterator that returns valid column elements, or a null
+ * replacement value for null elements and validity of the element in a pair
+ */
+template <typename Element>
+auto make_null_replacement_pair_iterator(column_device_view const& column,
```

Comment:
Is there a use case for a null replaced pair iterator?

Reply:
Yes. There is an example use case in unit test.

Reply:
Let me rephrase. Is there an actual _need_ for it?

Reply:
not exactly. It is needed for the example use case in unit test. But the example use case can be modified to not to use this iterator at all. 
should I remove and update the unit test?

Reply:
I generally don't like adding functionality unless there is a clear need for it, because every new feature adds to the cost of maintenance and it's a bummer to have to maintain things that aren't used.

Reply:
To be explicit: please remove it for now @karthikeyann .

Reply:
removed.